lib/devos/mkFlake.nix

```diff
@@ -0,0 +1,198 @@
+
+              # add backports from nixos'
+              "{$name}".backports = # TODO: pros & cons of namespacing with `backports`
+                let
+
+                  nixosPkgs' = import nixos' {
+                    inherit system config;
+                    overlay = [ ];
+                  };
+
+                in backportPkgs nixosPkgs' final prev
+            }
+
+        ]
+
+      # produces `packages`, `legacyPackages` & `devShell` - but also a multi pkgs
+      outputs' = flake-utils.lib.simpleFlake' { # TODO: re-export pkgs in fork
+        inherit
+          self
+          name
+          systems
+          overlay
+          shell
+          config
+          ;
+        preOverlays = preOverlays_;
+        nixpkgs = nixos;
+      };
+      outputs = outputs'.flake;
+      pkgs = outputs'.pkgs;
+
+      # re-exports according to the devos sharing model
+      exports = {
+        inherit
+          nixosModules
+          homeModules
+          # devShellModules
+          overlay
+          ;
+      };
+
+      configs = {
+        nixosConfigurations = let
+
+          specialArgs =
+            {
+              suites = suites.system;
+              modulesPath' = "${nixos'}/nixos/modules";
+            }
+            // extraSpecialArgs;
+
+          modules' =
+            builtins.attrValues self.nixosModules
+            ++ extraNixosModules;
+
+          hmSpecialArgs' =
+            {
+              suites = suites.user;
+            };
+            // extraHomeSpecialArgs;
+
+          hmModules' =
+            builtins.attrValues self.homeModules
+            ++ extraHomeModules;
+
+          modules = [
+
+            {
+              _module.args = {
+                inherit self;
+              };
+            }
+
+            {
+              nixpkgs.pkgs = lib.mkDefault pkgs.${config.nixpkgs.system};
+              hardware.enableRedistributableFirmware = lib.mkDefault true;
+            }
+
+            {
+              home-manager = {
+                useGlobalPkgs = true;
+                useUserPackages = true;
+
+                extraSpecialArgs = hmSpecialArgs';
+                sharedModules = hmModules';
+              };
+            }
+   
+            {
+              nix.extraOptions = ''
+                experimental-features = flakes nix-command ca-references ca-derivations
+              '';
+              nix.registry = {
+                self.flake = inputs.self;
+                nixos.flake = inputs.nixos;
+                override.flake = inputs.override;
+              };
+            }
+
+            { config, ... }: {
+              system.configurationRevision = lib.mkIf (self ? rev) self.rev;
+            }
+
+            { modulesPath', ... }: {
+              disabledModules = backportModules ++ disabledModules;
+              imports = let
+                f = path: "${modulesPath'}/${path}"
+              in map f backportModules;
+            };
+
+          ] ++ modules';
```

Comment:
I just put them here without any better naming or factoring because thei partly invoke self or similar stuff.
Those can be exquisitly put into a `./lib/modules/` folder, but I wasn't yet bothered with figuring out how to expose it's deps in a consistent way without just passing everything to `modules` and be a little more differential about what we pass along.